src/Board.cpp

```diff
@@ -0,0 +1,207 @@
+#include "Board.h"
+
+/**
+ * @brief Construct a new Board:: Board object
+ *
+ * @param board_string
+ */
+Board::Board(const string &board_string): m_isWhiteTurn(true) {
```

Comment:
Instead of initializing members inside the constructor body, consider using a member initializer list to initialize them. This can improve code readability and potentially improve performance. For example: Board::Board(const string &board_string)
    : m_isWhiteTurn(true), m_kingPos{{-1, -1}, {-1, -1}}